### ISSUE TITLE
Handle time range in the quickwit collector

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-inc/quickwit#123]
The time range needs to be used to prune splits, but also to filter documents befroe they reach the collector.
This part has not been implemented yet.

**Comment by evanxg852000** [comment]
@fulmicoton We already have `tantivy::collector::FilterCollector` appropriate for this. However, FilterCollector wraps a collector. 
Options I currently see:
- change `leaf_search` interface to accept a generic `C: Collector` or `&dyn Collector` instead of the QuickwitCollector. So that we can wrap QuickwitCollector in FilterCollector. This involves cascading interface change and I am feeling like there is a reason the interface was designed to take QuickwitCollector.
- implement the filtering of docs  inside QuickwitCollector (no favorable since we already have it in tantivy)
- Make QuickwitCollector an enum (restricting ourselves to enum variant)

in the first option, `make_collector` will have something like
```rust
let inner_collector = QuickwitCollector{...};
let timestamp_field_opt: Option<Field> = doc_mapper.timestamp_field()?;
    let collector = if let Some(timestamp_field) = timestamp_field_opt {
        let time_filter_predicate =
            make_time_filter_predicate(search_request.start_timestamp, search_request.end_timestamp);

        Box::new(FilterCollector::new(timestamp_field,time_filter_predicate,  inner_collector)) as Box<dyn Collector<Child=QuickwitSegmentCollector, Fruit=LeafSearchResult>>
    } else {
        Box::new(inner_collector) as Box<dyn Collector<Child=QuickwitSegmentCollector, Fruit=LeafSearchResult>>
    };

```

**Comment by fulmicoton** [comment]
All your analysis is correct.
I prefer the method you describe in the point 2)

The reason for this is that we need to handle the warmup part.
We can  leave to the QuickwitCollector to tell us all of the fastfields that are required to run the query.

The code duplication with the FilterCollector in tantivy is super minor.

**Comment by fulmicoton** [comment]
I just saw @fmassot  comment. We need the timestamp field to be a fast field.

**Comment by fmassot** [comment]
@fulmicoton sorry my comment was a bit misleading, using fast field for timestamp is needed for obvious reason. 
I was just broadening the subject because the user will be able to express range queries on several fields, not only on the timestamp field.

Example : 
- we have price field with a fast field on it
- the user wants to filter on it so it would be nice to use a collector for that.

Another example but with no fastfield:
- we have price field but with no fast field, it is only indexed
- the user wants to filter on it, it is handled in tantivy by a range query but it is expensive

**Comment by fmassot** [comment]
closed in #125
